examples/integrations/hf_trainer_api/det_callback.py

```diff
@@ -176,9 +189,17 @@ def on_epoch_end(
                 self.current_op.report_progress(state.epoch)
 
             if round(state.epoch) >= self.current_op.length:
```

Comment:
Modulo looking into this `round` call as we discussed, looks good to me.  Nice catch.